### PR TITLE
Transform REST server into Tokio service

### PR DESCRIPTION
**Pull request by CodeSandwich**
Closes https://github.com/input-output-hk/jormungandr/issues/913. Actix-web is monolithic, the only way to run it is in custom thread. This PR wraps it in a future making it controllable as a regular Tokio task:

- finish future when server stops (causing stopping the whole node)
- stop server when future is dropped (because other task has finished or crashed and whole node is being stopped)

### CODE REVIEW

File: jormungandr/src/rest/server/mod.rs [hunk 1]
```diff
@@ -10,40 +10,88 @@ use actix_web::{
...
+            Err(_) => Ok(Async::Ready(())),
+            Ok(ok) => Ok(ok),
+        }
     }
+}
+
+impl Drop for Server {
+    fn drop(&mut self) {
+        self.stopper.stop();
+        let _ = self.wait();
```
Comment (NicolasDP) [inline jormungandr/src/rest/server/mod.rs]: not sure about that though...

Comment (NicolasDP) [reply jormungandr/src/rest/server/mod.rs]: If there is a deadlock in there this is risky.

Comment (CodeSandwich) [reply jormungandr/src/rest/server/mod.rs]: Me neither. I've decided that it's better than having REST server randomly killed in the middle of network operation. Timeout would be the perfect solution, but at that point Tokio runtime is already down and it's not easy without its reactor.

Comment (mzabaluev) [reply jormungandr/src/rest/server/mod.rs]: Wait, what? Why do you need to do this? It should be OK to drop "disgracefully", that's how cancellation of everything works in futures.

Comment (mzabaluev) [reply jormungandr/src/rest/server/mod.rs]: If there is a graceful shutdown, it should be done with sending a `Shutdown` message to the task.

Comment (CodeSandwich) [reply jormungandr/src/rest/server/mod.rs]: :man_facepalming: that's so obvious!

**Pull request merged**